core/domain/user_jobs_one_off.py

```diff
@@ -73,6 +72,73 @@ def reduce(key, version_and_exp_ids):
                     edited_exploration_ids))
 
 
+class UserLanguageAuditOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for auditing the default language preferences of users."""
+
+    _LANGUAGE_TO_RESET_KEY = 'WOULD_RESET'
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        """Return a list of datastore class references to map over."""
+        return [user_models.UserSettingsModel]
+
+    @staticmethod
+    def map(item):
+        """Implements the map function for this job."""
+        if item.preferred_site_language_code in ['hu', 'mk', 'sv', 'tr']:
+            affected_users_key = '%s.%s' % (
+                UserLanguageAuditOneOffJob._LANGUAGE_TO_RESET_KEY,
+                item.preferred_site_language_code)
+            yield (
+                affected_users_key, {
+                    'language_code': item.preferred_site_language_code,
+                    'user_id': item.id
+                })
+        yield (item.preferred_site_language_code, 1)
+
+    @staticmethod
+    def reduce(key_or_language_code, stringified_values):
+        """Implements the reduce function for this job."""
+        if key_or_language_code.startswith(
+                UserLanguageAuditOneOffJob._LANGUAGE_TO_RESET_KEY):
+            affected_users = [
+                ast.literal_eval(stringified_affected_user)
+                for stringified_affected_user in stringified_values]
+            language_code = affected_users[0]['language_code']
+            yield 'Users affected by a reset of %s: %s' % (
+                language_code,
+                [affected_user['user_id'] for affected_user in affected_users])
+        else:
+            yield '%d users are using language: %s' % (
+                len(stringified_values), key_or_language_code)
+
+
+class UserLanguageResetOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for reseting the default language preferences of users."""
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        """Return a list of datastore class references to map over."""
+        return [user_models.UserSettingsModel]
+
+    @staticmethod
+    def map(item):
+        """Implements the map function for this job."""
+        if item.preferred_site_language_code in ['hu', 'mk', 'sv', 'tr']:
```

Comment:
list is used twice in this file, might be worth pulling out into a constant to avoid (theoretical) skew